src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -266,4 +436,97 @@ private void publishTypeMessage(File file, String msg) {
         currentTypeInfo.put(
                 com.google.common.collect.Range.closed(start, end), typeInfoForPosition);
     }
+
+    @Override
+    public CompletableFuture<List<Either<Command, CodeAction>>> codeAction(
```

Comment:
This is for refactoring support, right? Can you split those changes off into a separate PR?

Reply:
Sure.